scripts/release.sh

```
@@ -0,0 +1,278 @@
+CHANGELOG_VERSION="${VERSION#v}" # v2.14.0 → 2.14.0
+CHANGELOG_HEADER="## [$CHANGELOG_VERSION]"
+
+if ! grep -q "^## \[$CHANGELOG_VERSION\]" CHANGELOG.md; then
+    die "CHANGELOG.md missing section '$CHANGELOG_HEADER'. Add release notes first."
+fi
+log_ok "CHANGELOG.md contains $CHANGELOG_HEADER section"
+
+# ─── Validation 8: CI status on HEAD (optional) ────────────────────────────
+if [[ "$SKIP_CI_CHECK" != true ]]; then
+    if command -v gh >/dev/null 2>&1; then
+        CI_STATE="$(gh pr list --head "$CURRENT_BRANCH" --state merged --limit 1 --json number --jq '.[0].number // ""' 2>/dev/null || echo "")"
```

Shellcheck (0.11.0)

[warning] CI_STATE appears unused. Verify use (or export if used externally).

(SC2034)